examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+                setSelectedSlugFieldId(fieldId)
+            } else if (!isEnabling && fieldId === slugFieldId) {
+                // When disabling the current slug field, find next available one
+                const nextSlugField = possibleSlugFields.find(f => f.id !== fieldId)
+                setSelectedSlugFieldId(nextSlugField?.id ?? null)
+            }
+
+            return updatedDisabledFieldIds
+        })
+    }
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+
+        if (!selectedSlugFieldId) {
+            framer.notify("Slug field is required", {
+                variant: "error",
+            })
+            return
+        }
+
+        try {
+            setIsSyncing(true)
+            await onImport(
+                collection,
+                dataSource,
+                fields.filter(field => !disabledFieldIds.has(field.id)),
+                selectedSlugFieldId
+            )
+            await framer.closePlugin(`Synchronization successful`, {
+                variant: "success",
+            })
+        } catch (error) {
+            console.error(error)
+            framer.notify(`Failed to sync collection ${dataSource.id}`, {
+                variant: "error",
+            })
+        } finally {
+            setIsSyncing(false)
+        }
+    }
+
+    useLayoutEffect(() => {
+        framer.showUI({
+            width: UI_DEFAULTS.MAPPING_WIDTH,
+            height: UI_DEFAULTS.MAPPING_HEIGHT,
+            minWidth: UI_DEFAULTS.MAPPING_WIDTH,
+            minHeight: UI_DEFAULTS.MAPPING_HEIGHT,
+            resizable: true,
+        })
+    }, [])
+
+    return (
+        <main className="mapping">
+            <hr className="sticky-top" />
+            <form onSubmit={handleSubmit}>
+                <label className="slug-field" htmlFor="slugField">
+                    Slug Field
+                    <select
+                        name="slugField"
+                        className="field-input"
+                        value={selectedSlugFieldId ?? ""}
+                        onChange={event => setSelectedSlugFieldId(event.target.value)}
+                        required
+                    >
+                        {possibleSlugFields.map(field => {
+                            if (!field) return null
+                            return (
+                                <option key={field.id} value={field.id}>
+                                    {field.name}
+                                </option>
+                            )
+                        })}
+                    </select>
+                </label>
+
+                <div className="fields">
+                    <span className="column-span-2">Column</span>
+                    <span>Field</span>
+                    {fields.map((field, i) => (
+                        <FieldMappingRow
+                            key={field.id || i}
```

Comment:
Is it possible for the field `id` to not exist? If not, I would remove the index fallback. If you do need the index fallback I would rename `i` → `index`